### PR TITLE
Improve documentation of k8s storage configuration

### PR DESCRIPTION
**Pull request by moleksyuk**
Improved documentation with examples how to configure `Namerd` with `k8s` storage.

### CODE REVIEW

**Comment by wmorgan** [comment]
Thanks for this, @moleksyuk!

**Comment by moleksyuk** [comment]
@wmorgan, np :)

All examples based on real setup ;)

File: namerd/docs/storage.md [hunk 1]
```diff
@@ -23,19 +23,164 @@ Stores the dtab in memory.  Not suitable for production use.
...
   same namespace in which namerd is running. (default: "default")
 
+*Example:*  
+```yaml
+  storage:
+    kind: io.l5d.k8s
+    experimental: true
+```
+
+*Complete example of `Namerd` configuration with `k8s` storage and exposed 2 serivces for sync with `Linkerd` and `Namerd API`:*
```
Comment (esbie) [inline namerd/docs/storage.md]: s/serivces/services

File: namerd/docs/storage.md [hunk 2]
```diff
@@ -23,19 +23,164 @@ Stores the dtab in memory.  Not suitable for production use.
...
-* *port* -- Optional. The port used to connect to the Kubernetes API. (default: 443)
-* *tls*  -- Optional. Whether to connect to the Kubernetes API using TLS. (default: true)
-* *tlsWithoutValidation* -- Optional. Whether to disable certificate checking against the Kubernetes
+Stores the dtab with the Kubernetes master via the ThirdPartyResource APIs. 
+
+**Prerequisites**  
+1. Requires a cluster running Kubernetes 1.2+.  
+2. `kubectl` client 1.3+ is installed.  
+3. [ThirdPartyResource](http://kubernetes.io/docs/api-reference/extensions/v1beta1/definitions/#_v1beta1_thirdpartyresource) feature should be enabled in Kubernetes cluster.  
+  1. **How to check ThirdPartyResource is enabled**
```
Comment (esbie) [inline namerd/docs/storage.md]: This list has only one element. I would remove this line altogether

**Comment by esbie** [comment]
This example is extremely useful! It feels more guide-like than reference-like, so I think we should add it as part of the existing k8s guide https://linkerd.io/doc/0.7.4/k8s/

**Comment by adleong** [comment]
(The circleci failure appears to just be a test timing out.  Feel free to ignore it.)

File: namerd/docs/storage.md [hunk 3]
```diff
@@ -23,19 +23,164 @@ Stores the dtab in memory.  Not suitable for production use.
...
+        app: namerd
+    spec:
+      dnsPolicy: ClusterFirst
+      volumes:
+        - name: namerd-config
+          configMap:
+            name: namerd-config
+      containers:
+        - name: namerd
+          image: buoyantio/namerd:0.7.4
```
Comment (olix0r) [inline namerd/docs/storage.md]: Does anyone have suggestions wrt how we should deal with tags like this in our docs? We don't have a great way to keep these up to date...

File: namerd/docs/storage.md (hunk 1 above)
Comment (moleksyuk) [inline namerd/docs/storage.md]: Fixed

File: namerd/docs/storage.md (hunk 2 above)
Comment (moleksyuk) [inline namerd/docs/storage.md]: Replaced 1. to '*'. I think this "How to check..." should be as a sub-part of item `3. [ThirdPartyResource](http://kubernetes.io/docs/api-reference/extensions/v1beta1/definitions/#_v1beta1_thirdpartyresource) feature should be enabled in Kubernetes cluster.` and a list is the one possible way to visualize it in such way.

What do you think @esbie ?

File: namerd/docs/storage.md (hunk 3 above)
Comment (moleksyuk) [inline namerd/docs/storage.md]: @olix0r good point 👍 

What about such proposal?
```yaml
containers:
+        - name: namerd
+          image: buoyantio/namerd:<version> # specify required version or remove to use the latest
```

**Comment by moleksyuk** [comment]
@esbie 

>This example is extremely useful! It feels more guide-like than reference-like, so I think we should add it as part of the existing k8s guide https://linkerd.io/doc/0.7.4/k8s/

Later on (probably with other pull request) I'll prepare complete guide how to configure Linkerd+Namerd in Kubernetes from real example and list all pitfalls and how to resolve them.

File: namerd/docs/storage.md (hunk 2 above)
Comment (esbie) [inline namerd/docs/storage.md]: sure, works for me

**Comment by esbie** [comment]
👍  thanks @moleksyuk

File: namerd/docs/storage.md (hunk 3 above)
Comment (olix0r) [inline namerd/docs/storage.md]: Yeah, I think that's the best we can do at the moment.

**Comment by moleksyuk** [comment]
@olix0r I've replaced version numbers with placeholders.

**Comment by adleong** [comment]
These docs have been restructured somewhat.  Can you merge or rebase onto master?

**Comment by moleksyuk** [comment]
@adleong done 👍

**Comment by adleong** [comment]
Thanks for this, @moleksyuk!

**Pull request merged**